wrappers/python/simtk/openmm/app/forcefield.py

```diff
@@ -1674,6 +1674,186 @@ def postprocessSystem(self, sys, data, args):
+        if None not in types:
+            self.types1.append(types[0][0])
+            self.types2.append(types[1][0])
+            self.emin.append(_convertParameterToNumber(parameters['emin']))
+            self.rmin.append(_convertParameterToNumber(parameters['rmin']))
+
+    def registerLennardJones(self, parameters):
+        self.lj_types.registerAtom(parameters)
+
+
+    @staticmethod
+    def parseElement(element, ff):
+        existing = [f for f in ff._forces if isinstance(f, LennardJonesGenerator)]
+        if len(existing) == 0:
+            generator = LennardJonesGenerator(ff, float(element.attrib['lj14scale']))
+            ff.registerGenerator(generator)
+        else:
+            # Multiple <LennardJonesForce> tags were found, probably in different files
+            generator = existing[0]
+
+        for LJ in element.findall('Atom'):
+            generator.registerLennardJones(LJ.attrib)
+        for Nbfix in element.findall('NBFixPair'):
+            generator.registerNBFIX(Nbfix.attrib)
+
+    def createForce(self, sys, data, nonbondedMethod, nonbondedCutoff, args):
+        """ Ewald and PME will be interpreted as CutoffPeriodic for the CustomNonbondedForce """
+
+        # We need a CustomNonbondedForce to implement the NBFIX functionality.
+        # First derive the lookup tables
+
+        lj_indx_list = [0 for atom in data.atoms]
+        lj_radii, lj_depths = [], []
+        num_lj_types= 0
+        lj_type_list = []
+        for i, atom in enumerate(data.atoms):
+            atype = data.atomType[atom]
+            values = self.lj_types.paramsForType[atype]
+            if lj_indx_list[i]: continue
+            num_lj_types +=1
+            lj_indx_list[i] = num_lj_types
+            rmin = values['sigma'] * 2**(1.0/6) / 2
+            ljtype = (rmin, abs(values['epsilon']))
+            lj_type_list.append(atype)
+            lj_radii.append(rmin)
+            lj_depths.append(abs(values['epsilon']))
+            for j in range(i+1, len(data.atoms)):
+                atype2 = data.atomType[data.atoms[j]]
+                if lj_indx_list[j] > 0: continue
+                if atype2 == atype:
+                    lj_indx_list[j] = num_lj_types
+                elif not (atype in self.types1) or (atype in self.types2):
+                    # only non NBFIX types can be compressed
+                    values = self.lj_types.paramsForType[atype2]
+                    ljtype2 = (values['sigma']* 2**(1.0/6) / 2, abs(values['epsilon']))
+                    if ljtype == ljtype2:
+                        lj_indx_list[j] = num_lj_types
+        # Now everything is assigned. Create the A- and B-coefficient arrays
+        acoef = [0 for i in range(num_lj_types*num_lj_types)]
+        bcoef = acoef[:]
+        for m in range(num_lj_types):
+            for n in range(num_lj_types):
+                namem = lj_type_list[m]
+                namen = lj_type_list[n]
+                if range(len(self.types1)):
```

Comment:
I gather that range() evaluates to false if its argument is 0, true otherwise?  That's... perhaps not the clearest way of writing this. :)  How about `if len(self.types1) > 0`?